### PR TITLE
Log which task created log entry

### PR DESCRIPTION
**Pull request by fkorotkov**

### CODE REVIEW

File: internal/http_cache/http_cache.go [hunk 1]
```diff
@@ -88,10 +95,13 @@ func checkCacheExists(w http.ResponseWriter, cacheKey string) {
 		TaskIdentification: cirrusTaskIdentification,
 		CacheKey:           cacheKey,
 	}
-	_, err := client.CirrusClient.CacheInfo(context.Background(), &cacheInfoRequest)
+	response, err := client.CirrusClient.CacheInfo(context.Background(), &cacheInfoRequest)
 	if err != nil {
 		w.WriteHeader(http.StatusNotFound)
 	} else {
+		w.Header().Set(CirrusHeaderCreatedBy, strconv.FormatInt(response.Info.CreatedByTaskId, 10))
```
Comment (edigaryev) [inline internal/http_cache/http_cache.go:102]: The other fields seem to be unused, do we really need them?

File: internal/executor/cache.go [hunk 2]
```diff
@@ -350,7 +351,8 @@ func UploadCache(executor *Executor, commandName string, cacheHost string, instr
 		// check if some other task has uploaded the cache already
 		response, _ := httpClient.Head(fmt.Sprintf("http://%s/%s", cacheHost, cache.Key))
 		if response != nil && response.StatusCode == http.StatusOK {
-			logUploader.Write([]byte(fmt.Sprintf("\nSome other task has already uploaded cache entry %s! Skipping upload...", cache.Key)))
+			createdByTaskId := response.Header.Get(http_cache.CirrusHeaderCreatedBy)
```
Comment (edigaryev) [inline internal/executor/cache.go:354]: Wouldn't this be always empty when uploading to cache servers other than the one built into the agent?

File: internal/http_cache/http_cache.go [hunk 3]
```diff
@@ -13,11 +13,18 @@ import (
...
+	"strconv"
 )
 
 var cirrusTaskIdentification *api.TaskIdentification
 
-const activeRequestsPerLogicalCPU = 4
+const (
+	activeRequestsPerLogicalCPU = 4
+
+	CirrusHeaderCreatedBy         = "CIRRUS_CREATED_BY"
```
Comment (edigaryev) [inline internal/http_cache/http_cache.go:24]: Consider using more HTTP-like header names such as `Cirrus-Created-By`.

File: internal/http_cache/http_cache.go (hunk 1 above)
Comment (fkorotkov) [reply internal/http_cache/http_cache.go:102]: Can remove. Just exposed all the info available. 🤷

File: internal/executor/cache.go (hunk 2 above)
Comment (fkorotkov) [reply internal/executor/cache.go:354]: Unlikely but will add a check for an empty header.

Comment (fkorotkov) [reply internal/executor/cache.go:354]: See 80422f4

File: internal/http_cache/http_cache.go (hunk 3 above)
Comment (fkorotkov) [reply internal/http_cache/http_cache.go]: See d06f275

File: internal/http_cache/http_cache.go (hunk 1 above)
Comment (fkorotkov) [reply internal/http_cache/http_cache.go]: See d06f275

**Review by edigaryev — APPROVED** [review]

**Pull request merged**